### PR TITLE
Add markdown formatter

### PR DESCRIPTION
**Pull request by joe-sharp**
Add a new formatter that outputs markdown. I copied the style from the existing HTML formatter so that after processing, they should be the same looking to the end user. I also added tests for this new formatter.

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by bbatsov** [comment]
What's your use case for such a formatter?

**Comment by joe-sharp** [comment]
> What's your use case for such a formatter?

I made it to be able to add comments to GitHub PRs in a nice format. An additional use case could be for use with static site generators such as Jekyll.

File: spec/fixtures/markdown_formatter/expected.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
+# RuboCop Inspection Report
+
+3 files inspected, 22 offenses detected:
+
+### app/controllers/application_controller.rb - 1 offense
+  **Line # 1 - convention:** Style/FrozenStringLiteralComment: Missing frozen string literal comment.
```
Comment (bbatsov) [inline spec/fixtures/markdown_formatter/expected.md:6]: I'm thinking each offense should probably be a bullet-point with the code snippet nested underneath.

File: spec/fixtures/markdown_formatter/expected.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
+# RuboCop Inspection Report
+
+3 files inspected, 22 offenses detected:
+
+### app/controllers/application_controller.rb - 1 offense
```
Comment (bbatsov) [inline spec/fixtures/markdown_formatter/expected.md:5]: I'd probably put the offense count in parentheses here.

**Comment by bbatsov** [comment]
> I copied the style from the existing HTML formatter

I still made a couple of suggestions that I think would improve the readability of the generated output. You'll also have to mention this new formatter in the docs.

**Comment by bbatsov** [comment]
FYI - I'll be cutting a new release some time tomorrow, so we'll have to wrap this up today if you want it to make the release.

**Comment by joe-sharp** [comment]
@bbatsov thank you for the heads up about the release. I am not in a rush for this, but I do hope to pick up work on this today. Apologies for the delay!

**Comment by joe-sharp** [comment]
Thanks much for these suggestions, I think the report looks quite a bit nicer and definitely more readable!

**Review by bbatsov — APPROVED** [review]

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**